examples/resnet/resnet.chpl

```diff
@@ -0,0 +1,338 @@
+        const expansion: int;
+        var conv1: owned Conv2D(eltType);
+        var bn1: owned BatchNorm2D(eltType);
+        var conv2: owned Conv2D(eltType);
+        var bn2: owned BatchNorm2D(eltType);
+        var downsample: owned Sequential(eltType);
+        var emptyDownsample: bool;
+
+        proc init(type eltType = dtype, inplans: int, planes: int, stride: int){
+            super.init(eltType);
+            this.expansion = 1;
+            this.conv1 = new Conv2D(eltType, inplans, planes, kernel=3, stride=stride, bias=false);
+            this.bn1 = new BatchNorm2D(eltType, size=planes);
+            this.conv2 = new Conv2D(eltType, planes, planes, kernel=3, stride=1, bias=false);
+            this.bn2 = new BatchNorm2D(eltType, size=planes);
+            if stride != 1 || inplans != planes {
+                this.downsample = new Sequential(eltType,(
+                    new Conv2D(eltType, inplans, planes, kernel=1, stride=stride, bias=false)?,
+                    new BatchNorm2D(eltType, size=planes)?)
+                    , "downsample"
+                    );
+            } else {
+                this.downsample = new Sequential(eltType, true, "downsample");
+                this.emptyDownsample = true;
+            }
+
+            init this;
+            this.moduleName = "basicblock";
+
+            for (n,m) in moduleFields() {
+                addModule(n,m);
+            }
+        }
+
+        override proc forward(input: Tensor(eltType)): Tensor(eltType) {
+            var identity = input;
+            var outVal = this.conv1(input);
+            outVal = this.bn1(outVal);
+            outVal = outVal.relu();
+            outVal = this.conv2(outVal);
+            outVal = this.bn2(outVal);
+            if !this.emptyDownsample {
+                identity = this.downsample(identity);
+            }
+            outVal = outVal + identity;
+            outVal = outVal.relu();
+            return outVal;
+        }
+    }
+
+
+
+    class Bottleneck: Module(?) {
+        const expansion: int;
+        var conv1: owned Conv2D(eltType);
+        var bn1: owned BatchNorm2D(eltType);
+        var conv2: owned Conv2D(eltType);
+        var bn2: owned BatchNorm2D(eltType);
+        var conv3: owned Conv2D(eltType);
+        var bn3: owned BatchNorm2D(eltType);
+        var downsample: owned Sequential(eltType);
+        var emptyDownsample: bool;
+
+
+        proc init(type eltType = dtype, inplanes:int, planes:int, stride:int) {
+            super.init(eltType);
+            this.expansion = 4;
+            var width = planes; // width = int(planes * (base_width / 64.0)) * groups but groups = 1 and base_width = 64
+            this.conv1 = new Conv2D(eltType, inplanes, width, kernel=1, stride=1, bias=false);
+            this.bn1 = new BatchNorm2D(eltType, size=width);
+            this.conv2 = new Conv2D(eltType, width, width, kernel=3, stride=stride, padding=1, bias=false); // we don't have padding implemented yet
```

Comment:
is this comment still true?

Reply:
The comment on line 182 is still true, the comment on line 185 is not, we do have `padding` implemented now. Thanks for the catch!